### PR TITLE
Improve testing infrastructure

### PR DESCRIPTION
**Pull request by FinalAngel**
- [x] Update documentation
- [x] Add new testing infrastructure
- [x] Adapt gulpfile.js
- [x] Test
- [x] documentation review by Daniele

### CODE REVIEW

File: docs/testing/services.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
+********
+Services
+********
+
+
+Travis
+======
+
+The `Travis <http://docs.travis-ci.com/>`_ configuration is farily
```
Comment (dhorelik) [inline docs/testing/services.rst]: `fairly` :wink:

**Comment by lory87** [comment]
:100:

File: .jshintrc [hunk 2]
```diff
@@ -3,7 +3,6 @@
     "esnext": true,
     "expr": true,
     "bitwise": true,
-    "camelcase": true,
```
Comment (vxsx) [inline .jshintrc]: :(

Comment (FinalAngel) [inline .jshintrc]: yeah let's have a funeral about that

File: docs/testing/unit_tests.rst [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+This folder is added when running **unit tests** either through ``gulp tests``,
+``gulp tests:unit`` or ``gulp tests:watch``. Coverage uses the
+`istanbul <https://gotwarlost.github.io/istanbul/>`_ tool to give you a nice
+UI for debugging. Just simply launch the `index` file in either one of the
+sub-folders generated. There can be as many sub-folders as clients connected
+to your runner.
+
+It's worth to mention that the success of your project does not depend on the
+tests or the percentage of your code coverage, but it will improve maintanance
```
Comment (vxsx) [inline docs/testing/unit_tests.rst]: maintenance

**Comment by vxsx** [comment]
LGTM

**Pull request merged**